hedera-node/docs/design/services/smart-contract-service/contract-accounts-nonces.md

```diff
@@ -43,6 +41,17 @@ The following is a table with general use cases and behavior for Ethereum and He
     - `contractCreateResult` if the top level transaction was `ContractCreate`
     - `contractCallResult` if the top level transaction was `ContractCall`
 
+### Migration Of Nonces For Existing Contracts
+
+- We need perform a migration for all existing contract nonces from Consensus Node state to Mirror Node e.g. similar to traceability migration performed in the past
```

Comment:
Highlight the migration logic interfaces and classes.
The existing traceability migration classes probably shouldn't be used or extended as it's approach had lots of crossovers with autorenew assignments

Reply:
@david-bakin-sl any suggestions here.
Simple approach similar to but not the same as traceability migration or a simpler different approach?

Reply:
Here I meant that we should create new class (I will need update the `New Classes` section below as well, sorry) that implements the `SystemTask` interface and wire it in the `TaskModule` which is how the `TraceabilityExportTask` is implemented. Is this approach interfering with autorenew assignments in some way?

Reply:
Okay. New class with new cleaner logic is fine.
The previous problem was that traceability had auto-renew logic mixed in which it shouldn't have.